src/client.cpp

```diff
@@ -542,21 +539,31 @@ void Client::step(float dtime)
 			num_processed_meshes++;
 			MeshUpdateResult r = m_mesh_update_thread.m_queue_out.pop_frontNoEx();
 			MapBlock *block = m_env.getMap().getBlockNoCreateNoEx(r.p);
+			MinimapMapblock *minimap_mapblock = NULL;
 			if(block) {
 				// Delete the old mesh
-				if(block->mesh != NULL)
-				{
-					// TODO: Remove hardware buffers of meshbuffers of block->mesh
+				if(block->mesh != NULL)	{
 					delete block->mesh;
 					block->mesh = NULL;
 				}
 
-				// Replace with the new mesh
-				block->mesh = r.mesh;
+				if(r.mesh)
+					minimap_mapblock = r.mesh->getMinimapMapblock();
+
+				if(r.mesh && r.mesh->getMesh()->getMeshBufferCount() == 0) {
+					delete r.mesh;
+					block->mesh = NULL;
+				} else {
+					// Replace with the new mesh
+					block->mesh = r.mesh;
+				}
 			} else {
 				delete r.mesh;
+				minimap_mapblock = NULL;
 			}
 
+			m_mapper->addBlock(r.p, minimap_mapblock);
```

Comment:
No i cant, otherwise when i turn minimap on i couldnt get data to render it.
Anyway that only insterts into std::map pos and pointer to MinimapMapblock.